### PR TITLE
Improve Arabic contextual analysis (lv_txt_ap.c)

### PR DESCRIPTION
**Pull request by alsaleem00**
### Description of the feature or fix

There are two proposed changes :
(1) there is an Arabic character that was not included in the (ap_chars_map) array. Arabs use, not frequently, this character to make the text appear longer. If it is not treated correctly, it will break the Arabic contextual analysis
(2) handling of letters (lam-alef) sequence. If letters alef (4 shapes, 0x0622/0x0623/0x0625/0x0627) comes after the letter lam (0x0644), the result will be a new single shape that combines both. In order not to break the code flow (current code is doing in-place editing), I created an new parallel array (ch_fin) that receives new contextual shapes. At the end I copied the resultant array back to ch_enc.

lv_old
lv_new

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update CHANGELOG.md
- [x] Update the documentation. No need since it supposed to be part of Arabic contextual analysis.

### CODE REVIEW

**Comment by alsaleem00** [comment]
Sorry I missed changing the title. It should be "Improve Arabic contextual analysis (lv_txt_ap.c)"

File: src/lv_misc/lv_txt_ap.c [hunk 1]
```diff
@@ -125,16 +126,35 @@ uint32_t _lv_txt_ap_calc_bytes_cnt(const char * txt)
     return chars_cnt + 1;
 }
 
+uint32_t _lv_txt_lam_alef(uint32_t ch_curr, uint32_t ch_next)
```
Comment (kisvegabor) [inline src/lv_misc/lv_txt_ap.c]: Can we change it to `static`?

**Review by kisvegabor — COMMENTED** [review]
Thank you!

I don't know the Arabic language but I believe the changes are correct. :slightly_smiling_face:

**Review by embeddedt — COMMENTED** [review]
Please make sure to use 4-spaces indentation. There appears to be some sections with 2-space indentation (presumably because of your editor). Thanks. :slightly_smiling_face:

File: src/lv_misc/lv_txt_ap.c (hunk 1 above)
Comment (alsaleem00) [reply src/lv_misc/lv_txt_ap.c]: yes. i will change

Comment (alsaleem00) [reply src/lv_misc/lv_txt_ap.c]: I changed the function to static, moved it to static section, adjusted return type to follow other code variable types and tested it.

**Comment by alsaleem00** [comment]
> 
> 
> Please make sure to use 4-spaces indentation. There appears to be some sections with 2-space indentation (presumably because of your editor). Thanks. slightly_smiling_face

I reviewed the code (both in my editor and Github's) and could not find other than 4-space indentation. I reviewed the file using "edit file".
Could you point which line has this space indentation difference?
My editor is also 4-space indentation.
Thanks.

**Comment by embeddedt** [comment]
There are several spots, but [this is one of them](https://github.com/lvgl/lvgl/pull/2062/files#diff-af60518962a120804cc74a265c3f605496aaeb513cb9edcff64516392df5cb99R171-R180). It's strange that you don't see it, unless GitHub is showing it differently in the diff.

**Comment by alsaleem00** [comment]
Yes, indeed. If I view the code using "edit file" all look good. However, when I view them in "preview mode" some (most!) indentations are two spaces.
I re-indented them. Please have a look.

**Review by embeddedt — APPROVED** [review]

**Comment by kisvegabor** [comment]
The changes look good, thank you!

Please add a line about the update in Changelog too.

**Comment by alsaleem00** [comment]
Hi. How do I reach the Changelog file?
This is the changes I would like to add:
Change brief:
- Improving Arabic contextual analysis : added hyphen processing and proper handling of lam-alef sequence.

Thanks.

**Comment by embeddedt** [comment]
@alsaleem00 It's in the root folder of the repository. If you click here...

image

... you will be taken to the root of your fork where you can edit more files on the same pull request.

Since I'm a maintainer of this repo I can edit pull requests, so I made the change for you this time.

Thank you for the contribution!

**Comment by alsaleem00** [comment]
Thanks. I am glad that I have added a little thing to this great library.
Keep the good work.

**Comment by kisvegabor** [comment]
Thank you very much for your contribution! :slightly_smiling_face:

**Pull request merged**